src/components/FormSelectInput.component.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import styled from "styled-components";
+
+import { defaultTheme } from "../theme";
+import { InputStyle } from "./styled";
+
+type MapElement = { id: number; name: string };
+interface State extends MapElement {
+  cities: Array<MapElement>;
+}
+export const FormSelectInput = ({
+  label,
+  register,
+  onFocus,
+  onBlur,
+  onChange,
+  loading,
+  data,
+  disabled,
+}: {
+  label: string;
+  register: any;
+  onFocus?: any;
+  onBlur?: any;
+  onChange?: (e: React.ChangeEvent) => void;
+  loading: boolean;
+  data: any;
+  disabled?: boolean;
+}) => {
+  return (
+    <Select
+      {...register}
+      onFocus={onFocus}
+      onChange={onChange ? onChange : register.onChange}
+      disabled={disabled}
+      defaultValue=""
+      onBlur={onBlur ? onBlur : register.onBlur}
+    >
+      <option value="">{loading ? "Loading..." : "Select an option"}</option>
+      {label === "Cities" ? (
+        !loading &&
```

Comment:
same comment about ternary operators, please use different variables or simple `if` instead

Reply:
This file has been completely refactored to not use ternary operators, thanks!